### ISSUE TITLE
Rule Proposal: no unused variables/methods in `setup` return

### ISSUE THREAD

**Issue by Fubinator** [vuejs/eslint-plugin-vue#1919]
**Please describe what the rule should do:**
From time to time, especially after refactorings, unused variables/methods may end up in the return of the `setup` function. I think it would be cool to have a rule that checks if all the things returned in the `setup` function are also used in the `template`.

**What category should the rule belong to?**
- [x] Enforces code style (layout)
- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
  <div>No variables here</div>
</template>

<script>
import { ref }  from "vue"

export default {
  setup() {
    const unusedVariable = ref("I'm unused")

    function unusedFunction() {
      console.log("I'm unused, too!")
    }
 
    return {
      unusedVariable,  // Should error, because it is not used in the template
      unusedFunction  // Should error, because it is not used in the template
    }
  }
}
</script>
```

I did not find a rule that enforces this and no issue that proposes this rule. If I missed it, feel free to close the issue.

**Comment by jacekkarczmarczyk** [comment]
https://eslint.vuejs.org/rules/no-unused-properties.html
